Lib/ldap/ldapobject.py

```diff
@@ -39,10 +39,34 @@
   text_type = str
 
 
+# See SimpleLDAPObject._bytesify_input
+_LDAP_WARN_SKIP_FRAME = True
+
 class LDAPBytesWarning(BytesWarning):
   """python-ldap bytes mode warning
   """
 
+def _raise_byteswarning(message):
+    """Raise LDAPBytesWarning
+    """
+
+    # Call stacks that raise the warning tend to be complicated, so
+    # getting a useful stacklevel is tricky.
+    # We walk stack frames, ignoring functions in uninteresting files,
+    # based on the _LDAP_WARN_SKIP_FRAME marker in globals().
+    stacklevel = 2
+    try:
+        getframe = sys._getframe
+    except AttributeError:
+        pass
+    else:
+        frame = sys._getframe(stacklevel)
+    # walk up the stacks until we leave the file
+    while frame and frame.f_globals.get('_LDAP_WARN_SKIP_FRAME'):
```

Comment:
frame is not defined in case ``sys._getframe`` does not exist. Can we rather wrap the whole block into a ``hasattr(sys, '_getframe')`` check?

Reply:
Oops, thanks for the catch!